pallene/parser.lua

```diff
@@ -22,6 +22,7 @@ function Parser:init(lexer)
     self.region_depth = 0     -- Are we inside a type annotation?
     self.type_regions = {}    -- Sequence of pairs. Ranges of type annotations in program.
     self.comment_regions = {} -- Sequence of pairs. Ranges of comments in the program.
+    self.max_params = 200
```

Comment:
Since this is a constant, I think  it would be better to define it as a toplevel local variable, instead of as an instance variable.

While we are at it, it might also be a good idea to add a comment next to this variable, describing why the limit exists:
* Lua has a hard time calling functions with close to 255 arguments
* Lua itself does not allow functions with more than 200 local vars